miio/vacuum.py

```diff
@@ -412,19 +471,50 @@ def disable_dnd(self):
         return self.send("close_dnd_timer", [""])
 
     @command(
-        click.argument("speed", type=int),
+        click.argument("speed", type=str),
     )
-    def set_fan_speed(self, speed: int):
+    def set_fan_speed(self, speed: [int, FanSpeed]):
         """Set fan speed.
 
         :param int speed: Fan speed to set"""
         # speed = [38, 60 or 77]
+        if isinstance(speed, FanSpeed):
+            try:
+                speed = int(speed.value)
+
+            except ValueError:
+                for k, v in self._fanspeedMap.items():
+                    if v == speed:
+                        speed = k
+                        break
+
         return self.send("set_custom_mode", [speed])
 
     @command()
     def fan_speed(self):
         """Return fan speed."""
-        return self.send("get_custom_mode")[0]
+        speed = self.send("get_custom_mode")[0]
+        try:
+            if hasattr(self, '_fanspeedMap'):
```

Comment:
This should be already defined when it gets this far, so no need for checking, right?